### ISSUE TITLE
[TypeScript] Embedding InspectorFields inside component no longer works

### ISSUE THREAD

**Issue by shaddockh** [AtomicGameEngine/AtomicGameEngine#1301]
It looks like a new version of the TS compiler changed the way it handles internal object literals.  Previously:

```typescript
class Foo extends Atomic.JSComponent {
   inspectorFields = {
      speed: 1.0
  };
...
}
```
would work and would get converted in the JS to ```this.inspectorFields = ...```.

Now, it gets converted to ```_this.inspectorFields = ...``` which means it's no longer seen by the inspector.

There are some options:
1. Do not allow the ```inspectorFields``` to reside in the component and instead define it outside the component, but within the same module.  Need to update all the examples and project templates to handle it this way.
2. Specifically look for ```_this.inspectorFields``` in addition to ```this.inspectorFields```
3. Make the process that looks for the inspector fields actually instantiate the component and use normal JS methods to grab the contents of the inspector fields 

I'm opting for either 2 or 3.  3 would be more robust to future changes, but may require more work.  The quick fix would be 2 and then follow up with 3 for the long term solution.

**Comment by JoshEngebretson** [comment]
Closed via #1303

**Comment by JoshEngebretson** [comment]
Reopened as not solved via #1303 I don't think...

**Comment by shaddockh** [comment]
From what I can tell, this is working correctly.  What are you seeing?

**Comment by rosshadden** [comment]
It's working fine for me, too.

**Comment by JoshEngebretson** [comment]
I read "quick fix" on #1303 and thought there was still some consideration being put to options 1/3, so didn't want to close it out.  If this is working, all for the close option :)

**Comment by shaddockh** [comment]
ok, yeah that makes sense.  I'll break out option 3 as a new issue, but that is going to take some thinking about.  There may be dragons.
